packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,568 @@
+  final Map<T, Widget> children;
+
+  /// The identifier of the widget that is currently selected.
+  ///
+  /// This must be one of the keys in the [LinkedHashMap] of [children].
+  /// If this attribute is null, no widget will be initially selected.
+  final T groupValue;
+
+  /// The callback that is called when a new option is tapped.
+  ///
+  /// This attribute must not be null.
+  ///
+  /// The segmented control passes the newly selected widget's associated key
+  /// to the callback but does not actually change state until the parent
+  /// widget rebuilds the segmented control with the new [groupValue].
+  ///
+  /// The callback provided to [onValueChanged] should update the state of
+  /// the parent [StatefulWidget] using the [State.setState] method, so that
+  /// the parent gets rebuilt; for example:
+  ///
+  /// ## Sample code
+  ///
+  /// ```dart
+  /// class SegmentedControlExample extends StatefulWidget {
+  ///   @override
+  ///   State createState() => new SegmentedControlExampleState();
+  /// }
+  ///
+  /// class SegmentedControlExampleState extends State<SegmentedControlExample> {
+  ///   final Map<int, Widget> children = const {
+  ///     0: const Text('Child 1'),
+  ///     1: const Text('Child 2'),
+  ///   };
+  ///
+  ///   int currentValue;
+  ///
+  ///   @override
+  ///   Widget build(BuildContext context) {
+  ///     return new Container(
+  ///       child: new SegmentedControl<int>(
+  ///         children: children,
+  ///         onValueChanged: (int newValue) {
+  ///           setState(() {
+  ///             currentValue = newValue;
+  ///           });
+  ///         },
+  ///         groupValue: currentValue,
+  ///       ),
+  ///     );
+  ///   }
+  /// }
+  /// ```
+  final ValueChanged<T> onValueChanged;
+
+  @override
+  _SegmentedControlState<T> createState() => _SegmentedControlState<T>();
+}
+
+class _SegmentedControlState<T> extends State<SegmentedControl<T>> {
+  T _selectedValue;
+  T _pressedValue;
+  final List<Widget> _gestureChildren = <Widget>[];
+
+  @override
+  void initState() {
+    super.initState();
+    for (T currentKey in widget.children.keys) {
+      if (currentKey == widget.groupValue) {
+        _selectedValue = currentKey;
+      }
+    }
+  }
+
+  void _onTapDown(T currentKey) {
+    setState(() {
+      _pressedValue = currentKey;
+    });
+  }
+
+  void _onTapUp(TapUpDetails event) {
+    setState(() {
+      _pressedValue = null;
+    });
+  }
+
+  void _onTapCancel() {
+    setState(() {
+      _pressedValue = null;
+    });
+  }
+
+  void _onTap(T currentKey) {
+    setState(() {
+      if (_selectedValue != currentKey) {
+        widget.onValueChanged(currentKey);
+      }
+    });
+  }
+
+  @override
+  void didUpdateWidget(SegmentedControl<T> oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    for (T currentKey in widget.children.keys) {
+      if (currentKey == widget.groupValue) {
+        _selectedValue = currentKey;
+      }
+    }
```

Comment:
ditto

Reply:
Right now if `widget.children` changes such that the old selected value no longer exists in the map, you leave `_selectedValue` pointing at that old value instead of null. is that intended?